intelmq/bots/collectors/xmpp/collector.py

```diff
@@ -21,48 +39,84 @@ def init(self):
             self.stop()
 
     def process(self):
-
         if self.xmpp is None:
-            self.xmpp = sleekxmpp.XMPPBot(self.parameters.xmpp_user + '@' + self.parameters.xmpp_server,
-                                self.parameters.xmpp_password,
-                                self.send_message,
-                                self.logger)
+            self.xmpp = XMPPClientBot(self.parameters.xmpp_user + '@'
+                                      + self.parameters.xmpp_server,
+                                      self.parameters.xmpp_password,
+                                      self.parameters.xmpp_room,
+                                      self.parameters.xmpp_room_nick,
+                                      self.parameters.xmpp_room_password,
+                                      self.logger)
             self.xmpp.connect(reattempt=True)
             self.xmpp.process()
 
-    def killbot(self):
-        self.xmpp.disconnect(wait=True)
-        self.logger.info("Disconnected")
-
+            # Add Handlers and register Plugins
+            self.xmpp.register_plugin('xep_0030')  # Service Discovery
+            self.xmpp.register_plugin('xep_0045')  # Multi-User Chat
+            self.xmpp.add_event_handler("message", self.log_message)
+            self.xmpp.add_event_handler("groupchat_message", self.log_message)
+
+    def stop(self):
+        self.xmpp.disconnect()
+        self.logger.info("Disconnected from xmpp")
+
+        super(XMPPCollectorBot, self).stop()
+
+    def log_message(self, msg):
+        self.logger.debug("Received Stanza: %r , from %r", msg['body'],
+                          msg['from'])
+        self.logger.info("Stanza received")
+
+        raw_msg = msg['body']
+
+        # Read msg-body and add as raw to a new report.
+        # now it's up to a parser to do the interpretation of the message.
+        if raw_msg:
+            report = Report()
+            report.add("raw", raw_msg)
+            self.send_message(report)
+
+
+class XMPPClientBot(sleekxmpp.ClientXMPP):
```

Comment:
Could you rename this to `XMPPClient`, as this class is not a bot? (also for output)

Also see https://github.com/certtools/intelmq/blob/master/intelmq/bin/intelmqctl.py#L298 (yes, it's not nice)